packages/manager/apps/pci-object-storage/src/domain/regions/regionFilter.spec.ts

```diff
@@ -0,0 +1,572 @@
+import { describe, expect, it } from 'vitest';
+import { TRegionAvailability } from '@ovh-ux/manager-pci-common';
+import {
+  filterAllowedRegions,
+  groupRegionsByContinent,
+  ProductAvailability,
+  ProjectRegion,
+  RegionFilterCriteria,
+} from './regionFilter';
+import { OBJECT_CONTAINER_OFFER_STORAGE_STANDARD } from '@/constants';
+
+describe('filterAllowedRegions', () => {
+  const mockRegions: TRegionAvailability[] = [
+    {
+      name: 'EU-WEST-1',
+      type: 'region-3-az',
+      enabled: true,
+      datacenter: 'EU-WEST',
+      continentCode: 'EU',
+      availabilityZones: [],
+    },
+    {
+      name: 'EU-WEST',
+      type: 'region',
+      enabled: true,
+      datacenter: 'EU-WEST',
+      continentCode: 'EU',
+      availabilityZones: [],
+    },
+    {
+      name: 'US-EAST-1',
+      type: 'localzone',
+      enabled: true,
+      datacenter: 'US-EAST',
+      continentCode: 'NA',
+      availabilityZones: [],
+    },
+    {
+      name: 'ASIA-1',
+      type: 'region',
+      enabled: false,
+      datacenter: 'ASIA',
+      continentCode: 'AS',
+      availabilityZones: [],
+    },
+    {
+      name: 'EU-CENTRAL',
+      type: 'region-3-az',
+      enabled: true,
+      datacenter: 'EU-CENTRAL',
+      continentCode: 'EU',
+      availabilityZones: [],
+    },
+  ];
+
+  const mockProductAvailability: ProductAvailability = {
+    plans: [
+      {
+        code: 'storage-standard',
+        regions: mockRegions,
+      },
+      {
+        code: 'storage-swift',
+        regions: mockRegions.slice(0, 3),
+      },
+      {
+        code: 'non-storage-plan',
+        regions: mockRegions,
+      },
+    ],
+  };
+
+  const mockProjectRegions: ProjectRegion[] = [
+    {
+      name: 'EU-WEST-1',
+      services: [{ name: 'storage-swift' }, { name: 'storage-standard' }],
+    },
+    {
+      name: 'US-EAST-1',
+      services: [{ name: 'storage-standard' }],
+    },
+  ];
+
+  describe('basic filtering', () => {
+    it('should return empty array when availability is null', () => {
+      const result = filterAllowedRegions(null, mockProjectRegions);
+      expect(result).toEqual([]);
+    });
+
+    it('should return empty array when plans is undefined', () => {
+      const result = filterAllowedRegions({}, mockProjectRegions);
+      expect(result).toEqual([]);
+    });
+
+    it('should return empty array when plans is empty', () => {
+      const result = filterAllowedRegions({ plans: [] }, mockProjectRegions);
+      expect(result).toEqual([]);
+    });
+
+    it('should filter only storage plans', () => {
+      const result = filterAllowedRegions(
+        mockProductAvailability,
+        mockProjectRegions,
+      );
+
+      // Should include regions from storage plans only
+      expect(result.length).toBeGreaterThan(0);
```

Comment:
Since you controls your inputs in a test the lenght should be a precise number.

```suggestion
      expect(result.length).toEqual(X);
```